### ISSUE TITLE
swipeable and onShow Not Working Together in tabs

### ISSUE THREAD

**Issue by alyeldegwy** [Dogfalo/materialize#4545]
When trying to use onShow will using swipeable as setting for the tabs
onShow not Working 

when i remove swipeable, onShow works fine am i doing something wrong or this is a BUG?

**Comment by fega** [comment]
Hello @alyeldegwy, please follow the [contributing guidelines](https://github.com/Dogfalo/materialize/blob/master/CONTRIBUTING.md), and fill the Issue form

thanks

**Comment by sandor** [comment]
Guys, I can second this bug.

This will not work:

`$('#tabs-swipe10').tabs({ 'swipeable': true, onShow: function(tab) { console.log(tab); } });`

This is working:

$('#tabs-swipe10').tabs({ onShow: function(tab) { console.log(tab); } });

Any workaround for this yet? Would be highly appreciated...

**Comment by lordplagus02** [comment]
Can I third this bug? Cuz I'm experiencing this too.

**Comment by tomscholz** [comment]
> Can I third this bug?
What do you mean?

**Comment by lordplagus02** [comment]
Well @sandor seconded the bug... and I was third... so... anyway point is I have almost the exact same code as he and it doesn't work. Also worth noting is _no error message is displayed_.

**Comment by DanielRuf** [comment]
@lordplagus02 can you provide a codepen to show this?

**Comment by sandor** [comment]
@DanielRuf – sure Daniel here are two codepens demonstrating the problem...

Works if we click on the tabs:
http://codepen.io/sandor/pen/OmMOBP

Will not work on swipe or on tab-klick:
http://codepen.io/sandor/pen/pPgdXw

This should work if possible also on swipe...

**Comment by tomscholz** [comment]
Works for me though

**Comment by tomscholz** [comment]
@sandor Do you get any errors?

**Comment by DanielRuf** [comment]
`swipeable` is a string?

**Comment by DanielRuf** [comment]
> Works for me though

Click the tabs and take a look at the console, no output on the second codepen.

**Comment by DanielRuf** [comment]
`onShow` is not called in combination with `swipeable`, see https://github.com/Dogfalo/materialize/blob/master/js/tabs.js#L185-L196

Bug or not?

**Comment by tomscholz** [comment]
Idk. Ask @acburst @Dogfalo

**Comment by DanielRuf** [comment]
No info on this behavior here: http://materializecss.com/tabs.html
http://materializecss.com/tabs.html#swipeable
http://materializecss.com/tabs.html#options

**Comment by sandor** [comment]
@tomscholz – nope nor errors on my side. Just no output to the console (you should use the console of safari and not the one of codepen)....

**Comment by tomscholz** [comment]
@sandor Yeah, ^^ I know. I was looking at the wrong pen. You are right. No output. **in devtools** ;)

**Comment by sandor** [comment]
@tomscholz – hehe, I know it is uncool to use safari ;-) but I like the look ;-)

**Comment by vpezeshkian** [comment]
So, what's the fix for this? Any hacks before the fix come out, I'm on 0.98.0
BTW updating to 0.98.2 will break my tabs which is `{swipeable: true}`

**Comment by DanielRuf** [comment]
> BTW updating to 0.98.2 will break my tabs which is {swipeable: true}

Is this another issue?

The change was done before 0.98.0 was released https://github.com/Dogfalo/materialize/commit/5adccdbfee9433568d0dfeacabe0e99c46a35daa#diff-3dd5119e1daf5898ff764a06ac0aca24L135

Which is the last version where it worked with swipeable?

Not sure how and if we can add the onShow call after https://github.com/Dogfalo/materialize/commit/5adccdbfee9433568d0dfeacabe0e99c46a35daa#diff-3dd5119e1daf5898ff764a06ac0aca24R166

**Comment by sandor** [comment]
guys, sorry for my ignorance, but I am a bit confused now. So 0.98.2 has a fix for this bug? Or does 0.98.2 breaking swipable tabs? Hard for a beginner like me checking all the possibilities :-)

**Comment by DanielRuf** [comment]
> guys, sorry for my ignorance, but I am a bit confused now. So 0.98.2 has a fix for this bug? Or does 0.98.2 breaking swipable tabs? Hard for a beginner like me checking all the possibilities :-)

No, still no fix.

**Comment by sandor** [comment]
Just a naive question here – would it be possible to introduce some callbacks for this (I think this is than also related to some other components that are using the slider) like in slider.js (http://idangero.us/swiper/api/#.WPpzOTZ3VW8) – something like onXXXX(swiper, event)... Not that I could code this. But I wish I could.

**Comment by DanielRuf** [comment]
We could copy it but as already mentioned, not sure if bug or how it should work. @Dogfalo 
 and @acburst are the maintainers who can tell more as this is not described in the docs

https://github.com/Dogfalo/materialize/issues/4545#issuecomment-295362079

**Comment by DanielRuf** [comment]
@sandor could you try something like

```JavaScript
    if (options.swipeable) {
        if ($tabs_content.length) {
            $tabs_content.carousel('set', index);
            $content_swipe = $($(this).find('li.tab a').get(index).hash); // add
            if (typeof(options.onShow) === "function") { // these
                options.onShow.call(this, $content_swipe); // lines
            } // here
        }
    }
```

**Comment by edinantox** [comment]
@DanielRuf i got TypeError: $(...).find(...).get(...) is undefined

**Comment by sandor** [comment]
@DanielRuf  Thanks Daniel ! – will give that a try tomorrow... Too much red wine for now :-)

**Comment by DanielRuf** [comment]
@edinantox did you edit the tabs.js file and recompile materialize.js or change it directly?

Would have to test it. Was just a fast guess.

**Comment by edinantox** [comment]
@DanielRuf  directly, but i'm trying other things

**Comment by DanielRuf** [comment]
Ok, seems we can directly use `$content`, missed that.

**Comment by edinantox** [comment]
@DanielRuf i already try to use $content, but no deal.

**Comment by DanielRuf** [comment]
Right, just the click event is caught there using `on`.

Will check that later tomorrow (saturday).

Not sure but there is probably another part for the swipe / drag event.

**Comment by DanielRuf** [comment]
I guess `onCycleTo` is where we have to add this call.

Would be cool if someone could help me testing. =)

**Comment by sandor** [comment]
@DanielRuf Hey Daniel, sorry for the late replay. Do you have a complete scrip that I could test? Would give it a try...

**Comment by DanielRuf** [comment]
Currently not. You can copy the raw dist/js/materialize.js, host it on https://gist.github.com, change/add the lines, copy and paste the gist URL at https://rawgit.com and use the development URL in the codepen instead of the CDN hosted Materialize script.

**Comment by sandor** [comment]
@DanielRuf Hey Daniel, I have done my best. I have never worked with gist and stuff like that, and I am not a developer. I was not able to create the rawgit url so I used the raw URL from the gist. I have re linked my original codepen here http://codepen.io/sandor/pen/pPgdXw with the gist file. I can confirm that now clicking on the tabs will give the desired info in the protocolle. Swiping will also cause the console to do "something" – but the result is to cryptic for a poor designer like me... Anyhow – nor error that I can see... But maybe you just have a look at the codepen/gist?

**Comment by DanielRuf** [comment]
I will check that tomorrow.

**Comment by vpezeshkian** [comment]
Sorry for the late input, I wasn't clear in my post that I'm talking about 2 different issues. I have materializedcss as module in dependencies and when upgrading it to latest `v0.98.2` my existing code breaks. Yes, tabs do not work as it should.

And the other issue which is this thread, those two options (`swipeable` and `onShow`) do not work together. Any fix yet?

**Comment by DanielRuf** [comment]
> And the other issue which is this thread, those two options (swipeable and onShow) do not work together. Any fix yet?

https://github.com/Dogfalo/materialize/issues/4545#issuecomment-296324429 and the rest of the thread

**Comment by DanielRuf** [comment]
Also https://github.com/Dogfalo/materialize/issues/4545#issuecomment-296324429

If this is another issue please open a new issue for it here at GitHub

**Comment by goodits** [comment]
Just leaving a +1 as I'd love to be able to use swipeable and onShow together.

**Comment by acburst** [comment]
Fixed in 8129f11

**Comment by newpen** [comment]
It's returning the wrong tab when swiping.
